collector-bitmovin-player/src/main/java/com/bitmovin/analytics/bitmovin/player/BitmovinSdkAdapter.java

```diff
@@ -264,13 +272,19 @@ public DRMInformation getDRMInformation() {
         @Override
         public void onSourceLoaded(SourceLoadedEvent sourceLoadedEvent) {
             Log.d(TAG, "On Source Loaded");
+            isVideoPlayed = false;
+            isVideoAttemptedPlay = false;
         }
     };
 
     private OnSourceUnloadedListener onSourceUnloadedListener = new OnSourceUnloadedListener() {
         @Override
         public void onSourceUnloaded(SourceUnloadedEvent sourceUnloadedEvent) {
             Log.d(TAG, "On Source Unloaded");
+            if (!isVideoPlayed && isVideoAttemptedPlay) {
+                stateMachine.setVideoStartFailedReason(VideoStartFailedReason.PAGE_CLOSED);
+                stateMachine.transitionState(PlayerState.EXITBEFOREVIDEOSTART, getPosition());
```

Comment:
When an error happens the player is also unloading the source. Thus, there will be two samples sent out with the videoStartFailed data. Another reason to not use this  method is because when `player.load(...)` is called, the `unload` method will be also invoked. Please try to use a dedicated event to detect when the app closes, e.g. the `onDestroy` (not sure if this works 100% of the time): https://developer.android.com/reference/android/app/Activity#ActivityLifecycle

Reply:
This `onDestry()` event is from entire app, so I'm not really sure how to pass this event up to `BitmovinAnalytics` (this might be possible by adding new method here) and up to the `BitmovinSdkAdapter` where are all the flags and timer.

I can see that player has it internal `onDestroyedListener` so maybe it can be used for case when app is closed?

Reply:
Added onDestroy event listener